### PR TITLE
Discord channel support (create/delete)

### PR DESCRIPTION
**Pull request by acikek**

### CODE REVIEW

File: src/main/java/com/denizenscript/ddiscordbot/commands/DiscordChannelCommand.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import net.dv8tion.jda.api.entities.Category;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.requests.restaction.ChannelAction;
+import org.bukkit.Bukkit;
+
+public class DiscordChannelCommand extends AbstractCommand implements Holdable {
+
+    public DiscordChannelCommand() {
+        setName("discordchannel");
+        setSyntax("discordchannel [id:<id>] (group:<group_id>) (channel:<channel_id>) [create/delete] (name:<name>) (description:<description>) (category:<category_id>)");
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/commands/DiscordChannelCommand.java]: This is two separate commands (create and delete). Or probably even a create command and a delete _mech_ considering delete takes effectively zero input.

File: src/main/java/com/denizenscript/ddiscordbot/commands/DiscordChannelCommand.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        setRequiredArguments(3, 6);
+    }
+    // <--[command]
+    // @Name discordchannel
+    // @Syntax discordchannel [id:<id>] (group:<group_id>) (channel:<channel_id>) [create/delete] (name:<name>) (description:<description>) (category:<category_id>)
+    // @Required 3
+    // @Maximum 6
+    // @Short Manages text channels on Discord.
+    // @Plugin dDiscordBot
+    // @Guide https://guide.denizenscript.com/guides/expanding/ddiscordbot.html
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/commands/DiscordChannelCommand.java]: I don't think the dDiscordBot guide page covers this at all, unless you're planning a followup guide PR? (imo redundant to do, better to just remove the guide link here)

**Comment by acikek** [comment]
- Changed `discordchannel` to `discordcreatechannel`
- Moved deletion to `DiscordChannelTag.delete` mech

File: src/main/java/com/denizenscript/ddiscordbot/objects/DiscordChannelTag.java [hunk 3]
```diff
@@ -541,5 +541,15 @@ public void adjust(Mechanism mechanism) {
             }
             ((ThreadChannel) channel).getManager().setLocked(mechanism.getValue().asBoolean()).submit();
         }
+
+        // <--[mechanism]
+        // @object DiscordChannelTag
+        // @name delete
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/objects/DiscordChannelTag.java:547]: you can't just delete the input line... it'd be `// @input None`

File: src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCreateChannelCommand.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+public class DiscordCreateChannelCommand extends AbstractCommand implements Holdable {
+
+    public DiscordCreateChannelCommand() {
+        setName("discordcreatechannel");
+        setSyntax("discordcreatechannel [id:<id>] [group:<group_id>] [name:<name>] (description:<description>) (category:<category_id>) (position:<#>)");
+        setRequiredArguments(3, 6);
+    }
+    // <--[command]
+    // @Name discordcreatechannel
+    // @Syntax discordcreatechannel [id:<id>] [group:<group_id>] [name:<name>] (description:<description>) (category:<category_id>) (position:<#>)
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCreateChannelCommand.java]: shouldn't have `_id` on `group`

File: src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCreateChannelCommand.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        Bukkit.getScheduler().runTaskAsynchronously(DenizenDiscordBot.instance, () -> {
+            try {
+                ChannelAction<TextChannel> action = group.getGuild().createTextChannel(name.asString());
+                if (description != null) {
+                    action = action.setTopic(description.asString());
+                }
+                if (category != null) {
+                    Category resultCategory = group.getGuild().getCategoryById(category.asString());
+                    if (resultCategory == null) {
+                        Debug.echoError("Invalid category!");
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCreateChannelCommand.java]: has to be scheduled onto main thread, and also include the `scriptEntry` reference as the first param

Comment (acikek) [reply src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCreateChannelCommand.java]: How do I accomplish the first, and what does the second mean?

Comment (mcmonkey4eva) [reply src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCreateChannelCommand.java]: look at `DiscordCreateThreadCommand`

Comment (acikek) [reply src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCreateChannelCommand.java]: I've pushed a working change, but I still don't know what the latter part of your initial comment means

**Pull request merged**